fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/ServerEntityEvents.java

```diff
@@ -84,4 +98,9 @@ public interface Load {
 	public interface Unload {
 		void onUnload(Entity entity, ServerWorld world);
 	}
+
+	@FunctionalInterface
+	public interface EquipmentChange {
+		void onChange(LivingEntity livingEntity, EquipmentSlot equipmentSlot, ItemStack previousStack, ItemStack currentStack);
```

Comment:
While the current injection point is problematic for this use case, would it be considered in-scope for this event or another event to potentially reject equipment changes? [Plasmid's `GameRuleType.MODIFY_ARMOR`](https://github.com/NucleoidMC/plasmid/blob/87c4bb18ae02f8d11eafd561a2421bdbbc60e3a0/src/main/java/xyz/nucleoid/plasmid/mixin/game/rule/ServerPlayNetworkHandlerMixin.java#L56-L96), for example, could be greatly simplified in such a case.

Reply:
I think this is out of scope for this PR but could definally be a good candidate for something in a new PR. I have only had a quick look but it may be better to also have a client side component for an event that prevents slot usage.